exporter/splunkhecexporter/client.go

```diff
@@ -650,6 +659,10 @@ func (c *client) start(ctx context.Context, host component.Host) (err error) {
 	}
 	url, _ := c.config.getURL()
 	c.hecWorker = &defaultHecWorker{url, httpClient, buildHTTPHeaders(c.config, c.buildInfo)}
+	c.heartbeater = newHeartbeater(c.config, getPushLogFn(c))
+	if c.heartbeater != nil {
+		c.heartbeater.initHeartbeat(c.buildInfo)
+	}
```

Comment:
If it's already part of the client.start() function, do we really need to separate `newHeartbeater` and `initHeartbeat`? I think they can be part of one function `newHeartbeater`

Reply:
that is fair. will move it inside